ggshield/utils/git_shell.py

```diff
@@ -95,6 +96,25 @@ def _git_rev_parse(option: str, wd: Path) -> Optional[str]:
     return _git_rev_parse_absolute(option=option, wd_absolute=wd.resolve())
 
 
+def _parse_from_git_url(url: str) -> str:
```

Comment:
This function is not really a parser. I would call it `simplify_git_url()` or `clean_git_url()`.